File: medium/cpp/487.cpp
// https://leetcode.com/problems/max-consecutive-ones-ii/description/

class Solution
{
public:
    int findMaxConsecutiveOnes(std::vector<int>& nums)
    {
        auto flipped = 0;
        auto res = 0;

        for (auto r = 0, l = 0; r < std::size(nums); r++)
        {
            if (nums[r] == 0)
            {
                flipped++;
            }

            while (flipped > 1)
            {
                if (nums[l++] == 0)
                {
                    flipped--;
                }
            }

            res = std::max(res, r - l + 1);
        }

        return res;
    }
};

File: medium/python/487.py
# https://leetcode.com/problems/max-consecutive-ones-ii/

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        flipped = 0
        res = 0
        l, r = 0, 0

        while r < len(nums):
            if nums[r] == 0:
                flipped += 1

            while flipped == 2:
                if nums[l] == 0:
                    flipped -= 1
                l += 1

            res = max(res, r - l + 1)
            r += 1

        return res
        